From b3624575ad7852b37ee45fdf4364ac35f3d637b0 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:56 -0400
Subject: [tools] perf: Do not try to call addr2line on non-binary files

Message-id: <1408462094-14194-128-git-send-email-jolsa@redhat.com>
Patchwork-id: 88130
O-Subject: [PATCH RHEL7.1 BZ1131394 127/325] perf tools: Do not try to call addr2line on non-binary files
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 58d91a0068694a5ba3efc99e88ce6b4b0dd0d085
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Wed Sep 11 14:09:29 2013 +0900

description
===========
No need to call addr2line since they don't have such information.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/sort.c b/tools/perf/util/sort.c
index b7e0ef0..d443593 100644
--- a/tools/perf/util/sort.c
+++ b/tools/perf/util/sort.c
@@ -259,9 +259,6 @@ static int hist_entry__srcline_snprintf(struct hist_entry *self, char *bf,
  if (!self->ms.map)
   goto out_ip;
 
- if (!strncmp(self->ms.map->dso->long_name, "/tmp/perf-", 10))
-  goto out_ip;
-
  path = get_srcline(self->ms.map->dso->long_name, self->ip);
  self->srcline = path;
 
diff --git a/tools/perf/util/srcline.c b/tools/perf/util/srcline.c
index 7e92cca..777f918 100644
--- a/tools/perf/util/srcline.c
+++ b/tools/perf/util/srcline.c
@@ -57,11 +57,17 @@ char *get_srcline(const char *dso_name, unsigned long addr)
 {
  char *file;
  unsigned line;
- char *srcline;
+ char *srcline = SRCLINE_UNKNOWN;
  size_t size;
 
+ if (dso_name[0] == '[')
+  goto out;
+
+ if (!strncmp(dso_name, "/tmp/perf-", 10))
+  goto out;
+
  if (!addr2line(dso_name, addr, &file, &line))
-  return SRCLINE_UNKNOWN;
+  goto out;
 
  /* just calculate actual length */
  size = snprintf(NULL, 0, "%s:%u", file, line) + 1;
@@ -73,6 +79,7 @@ char *get_srcline(const char *dso_name, unsigned long addr)
   srcline = SRCLINE_UNKNOWN;
 
  free(file);
+out:
  return srcline;
 }
 
-- 
1.7.1